docs/aws/audit/cloudfrontmonitoring/rules/cloudfront_distributions_https_remediation.mdx

```diff
@@ -1,6 +1,260 @@
+
+   ```sh
+   aws cloudfront create-distribution --distribution-config file://distribution-config.json
+   ```
+
+   or if updating an existing distribution:
+
+   ```sh
+   aws cloudfront update-distribution --id <DISTRIBUTION_ID> --distribution-config file://distribution-config.json
+   ```
+
+   In your `distribution-config.json`, ensure the `ViewerProtocolPolicy` is set correctly:
+
+   ```json
+   {
+     "ViewerCertificate": {
+       "CloudFrontDefaultCertificate": true,
+       "MinimumProtocolVersion": "TLSv1.2_2018"
+     },
+     "DefaultCacheBehavior": {
+       "ViewerProtocolPolicy": "redirect-to-https"
+     }
+   }
+   ```
+
+2. **Set Up Custom SSL Certificate:**
+   If you are using a custom domain, ensure you have an SSL certificate from AWS Certificate Manager (ACM) and associate it with your CloudFront distribution.
+
+   ```sh
+   aws cloudfront update-distribution --id <DISTRIBUTION_ID> --distribution-config file://distribution-config.json
+   ```
+
+   In your `distribution-config.json`, include the `ACMCertificateArn`:
+
+   ```json
+   {
+     "ViewerCertificate": {
+       "ACMCertificateArn": "arn:aws:acm:region:account-id:certificate/certificate-id",
+       "SSLSupportMethod": "sni-only",
+       "MinimumProtocolVersion": "TLSv1.2_2018"
+     }
+   }
+   ```
+
+3. **Ensure All Cache Behaviors Use HTTPS:**
+   Make sure all cache behaviors in your distribution are set to use HTTPS.
+
+   ```sh
+   aws cloudfront update-distribution --id <DISTRIBUTION_ID> --distribution-config file://distribution-config.json
+   ```
+
+   In your `distribution-config.json`, for each cache behavior, set `ViewerProtocolPolicy` to `redirect-to-https` or `https-only`:
+
+   ```json
+   {
+     "CacheBehaviors": [
+       {
+         "PathPattern": "/path",
+         "ViewerProtocolPolicy": "redirect-to-https"
+       }
+     ]
+   }
+   ```
+
+4. **Verify Distribution Configuration:**
+   After making changes, verify the configuration to ensure HTTPS is enabled.
+
+   ```sh
+   aws cloudfront get-distribution-config --id <DISTRIBUTION_ID>
+   ```
+
+   Check the output to confirm that `ViewerProtocolPolicy` is set to `redirect-to-https` or `https-only` and that the `ViewerCertificate` is properly configured.
+
+By following these steps, you can ensure that HTTPS is enabled on your CloudFront distributions using AWS CLI.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent the misconfiguration of not having HTTPS enabled on CloudFront distributions using Python scripts, you can follow these steps:
+
+1. **Set Up AWS SDK for Python (Boto3):**
+   - Ensure you have Boto3 installed. If not, install it using pip:
+     ```bash
+     pip install boto3
+     ```
+
+2. **Create a Boto3 Session:**
+   - Initialize a session with your AWS credentials.
+   ```python
+   import boto3
+
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>